language/compiler/ir-to-bytecode/syntax/src/syntax.rs

```diff
@@ -62,31 +62,49 @@ fn consume_token<'input>(
     Ok(())
 }
 
-fn parse_comma_list<'input, F, R>(
+fn parse_comma_list<'input, F, G, R>(
     tokens: &mut Lexer<'input>,
-    list_end_token: Tok,
+    is_end_of_list: G,
```

Comment:
Per my response to your comment in https://github.com/libra/libra/pull/1973, this approach with "is_end_of_list" is going to conflict with my current work to print the expected end-of-list token in error messages. I prefer the version of the code that I sent you earlier that always checks for and replaces ">>" when the end-of-list token is ">". Is there a reason you think that won't work? It seems simpler and more flexible to me, as long as we never expect to see ">>" in this context.